bottomless/src/replicator.rs

```diff
@@ -801,12 +872,12 @@ impl Replicator {
         if let Some(tombstone) = self.get_tombstone().await? {
             if let Some(timestamp) = Self::generation_to_timestamp(&generation) {
                 if tombstone.timestamp() as u64 >= timestamp.to_unix().0 {
-                    tracing::error!(
-                        "Restoration failed. Database '{}' has been tombstoned at {}.",
+                    bail!(
```

Comment:
Question to @MarinPostma : this is when we try to restore db when ie. namespace was deleted. How should we react? 

1. Ignore restore with just warning (current status quo)?
2. Return from restore with error (this PR)?